src/shared/credentials/defaultCredentialsFileProcessor.ts

```diff
@@ -0,0 +1,44 @@
+/*!
+ * Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+
+import { IniLoader } from 'aws-sdk'
+import { CredentialsFileReaderProcessor, Profile } from './credentialsFileProcessor'
+
+export class DefaultCredentialsFileProcessor implements CredentialsFileReaderProcessor {
+    public async getProfileNames(): Promise<string[]> {
+        // TODO: cache the file and attach a watcher to it
+        const iniLoader = new IniLoader()
+        const profiles = iniLoader.loadFrom({})
+
+        return Promise.resolve(Array.from(Object.keys(profiles)))
```

Comment:
You don't need to call `Promise.resolve`. Because the method is `async` the return value is automatically wrapped in a promise. Same for rest of this file.